src/components/Header/Header.jsx

```diff
@@ -1,23 +1,69 @@
-import React, { useContext, useState } from 'react';
+import React, { useContext, useState, useEffect } from 'react';
 import UserContext from '../../Context/UserContext';
 import FilterableList from '../FilterableList/FilterableList';
 import Button from '../Button/Button';
 import MenuList from '../MenuList/MenuList';
 import styles from './Header.scss';
+import axios from 'axios';
+import { useHistory } from 'react-router';
 
 function Header() {
   const [isVisible, setIsVisible] = useState(false);
-  const { capitals } = useContext(UserContext);
+  const [cities, setCities] = useState([]);
+  const [inputValue, setInputValue] = useState('');
+  const history = useHistory();
 
   const handleClick = () => {
     setIsVisible(prev => !prev);
   };
 
+  const context = useContext(UserContext);
+  const token = context.token !== '' && context.token;
+
+  useEffect(() => {
+    const cancelTokenSource = axios.CancelToken.source();
+    if (token !== false && inputValue !== '') {
+      try {
+        axios
+          .get(`https://pfa.foreca.com/api/v1/location/search/${inputValue}`, {
+            headers: {
```

Comment:
You can use either sessionStorage to save your token for example. Then you won't need to use Provider. ALso you may use axios request interceptors to add headers of your desire in on;y one place.

Reply:
My idea is to use Context and to store token and current city(in next task when we must show current city), because list page with city is flyout in the header. And if don's store current city in context or in other place I must lift state to home page.